internal/pkgmanager/ecosystem.go

```diff
@@ -11,10 +12,10 @@ import (
 // PkgManager represents how packages from a common ecosystem are accessed.
 type PkgManager struct {
 	ecosystem       pkgecosystem.Ecosystem
-	latestVersion   func(string) (string, error)
-	archiveURL      func(string, string) (string, error)
-	archiveFilename func(string, string, string) string
-	extractArchive  func(string, string) error
+	latestVersion   func(name string) (string, error)
+	archiveURL      func(name string, version string) (string, error)
+	archiveFilename func(name string, version string, downloadURL string) string
+	extractArchive  func(path string, outputDir string) error
```

Comment:
nit: go lets you avoid writting the same type over and over again in arg lists:

e.g. `func(name string, version string, downloadURL string) string` can be `func(name, version, downloadURL string) string`

Reply:
done